staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -354,7 +364,11 @@ func (d *Helper) evictPods(pods []corev1.Pod, evictionGroupVersion schema.GroupV
 			if err == nil {
 				returnCh <- nil
 			} else {
-				returnCh <- fmt.Errorf("error when waiting for pod %q in namespace %q to terminate: %v", pod.Name, pod.Namespace, err)
+				err = fmt.Errorf("error when waiting for pod %q in namespace %q to terminate: %w", pod.Name, pod.Namespace, err)
+				if d.OnPodDeletionOrEvictionFinished != nil {
```

Comment:
in case of timeout in `waitForDelete` we are not calling `OnPodDeletionOrEvictionFinished` inside `waitForDelete` function, so it needs to be handled here. The error returned from `waitForDelete` is a polling error, let me know if I understand it wrong.

Reply:
When this https://github.com/kubernetes/kubernetes/blob/4aecb151b8301a6fc69f63c08f640c7b2b4289fc/staging/src/k8s.io/kubectl/pkg/drain/drain.go#L433 happens, won't be  `OnPodDeletionOrEvictionFinished` executed twice?

Reply:
sorry, I missed it. I will fix it and also add unit tests.

Reply:
No worries, thank you.